test/mitmproxy/proxy/protocol/test_http2.py

```diff
@@ -862,3 +863,91 @@ def test_connection_terminated(self):
         assert connection_terminated_event.error_code == 5
         assert connection_terminated_event.last_stream_id == 42
         assert connection_terminated_event.additional_data == b'foobar'
+
+
+class TestRequestStreaming(_Http2Test):
+
+    @classmethod
+    def handle_server_event(cls, event, h2_conn, rfile, wfile):
+        if isinstance(event, h2.events.ConnectionTerminated):
+            return False
+        elif isinstance(event, h2.events.RequestReceived):
+            req = rfile.read(109)
+            assert req
+            h2_conn.close_connection(error_code=5, last_stream_id=42, additional_data=req[9:])
+            wfile.write(h2_conn.data_to_send())
+            wfile.flush()
+
+        return True
+
+    def _send_request(self,
```

Comment:
Please modify the existing `_send_request` function so we can change this with a new argument / config.